src/fides/api/ops/service/messaging/message_dispatch_service.py

```diff
@@ -411,6 +415,49 @@ def _mailgun_dispatcher(
         raise MessageDispatchException(f"Email failed to send due to: {Pii(e)}")
 
 
+def _twilio_email_dispatcher(
+    messaging_config: MessagingConfig,
+    message: EmailForActionType,
+    to: Optional[str],
+) -> None:
+    """Dispatches email using twilio sendgrid"""
+    if not to:
+        logger.error("Message failed to send. No email identity supplied.")
+        raise MessageDispatchException("No email identity supplied.")
+    if not messaging_config.details or not messaging_config.secrets:
+        logger.error(
+            "Message failed to send. No twilio email config details or secrets supplied."
+        )
+        raise MessageDispatchException(
+            "No twilio email config details or secrets supplied."
+        )
+
+    try:
+        sg = sendgrid.SendGridAPIClient(
+            api_key=messaging_config.secrets[
+                MessagingServiceSecrets.TWILIO_API_KEY.value
+            ]
+        )
+        from_email = Email(
+            messaging_config.details[MessagingServiceDetails.TWILIO_EMAIL_FROM.value]
```

Comment:
a little bit of an edge case, but we may want to specifically handle a possible `KeyError` here, since i don't think there's any previous validation that ensures we've got a `TWILIO_EMAIL_FROM` key in our `details` dict (given that `details` field on the top-level `MessagingConfigRequest` schema can also take the `MessagingServiceDetailsMailgun` keys). 

i tested this edge case locally, and we'll error out, but the error messages are a bit hard to understand, so i think we may be able to handle it better if we specifically catch the `KeyError`. specifically, i tested by:
- providing a key in the `details` nested object that is used for mailgun, not twilio, e.g. 
```
{
    "key": "twilio_email_key",
    "name": "twilio_email",
    "service_type": "twilio_email",
    "details": {
        "domain": "your_sender_email_in_sendgrid@email.com"
    }
}
```
(note the **domain** key within `details`)
- adding secrets and executing a privacy request
- errors in my logs looked like this -- not _horrible_ but also not super clear:
```
fides-fides-1  | 2023-01-06 18:30:45.406 [ERROR] (message_dispatch_service:_twilio_email_dispatcher:457): Email failed to send: 'twilio_email_from'
fides-fides-1  | 2023-01-06 18:30:45.406 [ERROR] (privacy_request_endpoints:_create_privacy_request:1662): MessageDispatchException: Email failed to send due to: 'twilio_email_from'
```

Reply:
alternatively, instead of catching the `KeyError` here, you could just add an extra check for this specific key into the conditional on L427 that you've already got - that may read a bit cleaner.

Reply:
nice, i like that a lot!